packages/@aws-cdk/aws-iotevents/test/detector-model.test.ts

```diff
@@ -137,6 +137,63 @@ test('can set multiple events to State', () => {
   });
 });
 
+test('can set states with transitions', () => {
+  // WHEN
+  const firstState = new iotevents.State({
+    stateName: 'firstState',
+    onEnter: [{
+      eventName: 'test-eventName',
+      condition: iotevents.Expression.fromString('test-eventCondition'),
+    }],
+  });
+  const secondState = new iotevents.State({
+    stateName: 'secondState',
+  });
+
+  firstState.transitionTo({
+    eventName: 'firstToSecond',
+    nextState: secondState,
+    condition: iotevents.Expression.fromString('test-eventCondition-12'),
+  });
+  secondState.transitionTo({
+    eventName: 'secondToFirst',
+    nextState: firstState,
+    condition: iotevents.Expression.fromString('test-eventCondition-21'),
+  });
```

Comment:
This test confirm that no ｄuplicate states were created if there is circular reference.